src/solver/loss/forward_hinge_embedding_loss.cpp

```diff
@@ -0,0 +1,163 @@
+
+namespace miopen {
+
+namespace solver {
+
+namespace loss {
+
+bool HingeEmbeddingLossFwd::IsApplicable(
+    const ExecutionContext& /*context*/,
+    const miopen::loss::HingeEmbeddingLossFwdProblemDescription& problem) const
+{
+    if(problem.GetInputDesc().GetSize() > 5)
+        return false;
+    return true;
+}
+
+ConvSolution HingeEmbeddingLossFwd::GetSolution(
+    const ExecutionContext& context,
+    const miopen::loss::HingeEmbeddingLossFwdProblemDescription& problem) const
+{
+    std::ignore = context;
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype        = problem.GetOutputDesc().GetType();
+    auto in_dtype     = miopen::GetDataType(problem.GetInputDesc().GetType());
+    auto target_dtype = miopen::GetDataType(problem.GetTargetDesc().GetType());
+    auto size         = problem.GetInputDesc().GetElementSize();
+
+    const auto build_params = KernelBuildParameters{
+        {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+        {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+        {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+        {"IN_OUT_TYPE", in_dtype == "bfloat16" ? "ushort" : in_dtype},
+        {"TARGET_TYPE", target_dtype},
+        {"LOCAL_SIZE", LOCAL_SIZE},
+    };
+
+    /* Phase 1: Add loss kernel */
+    result.construction_params.push_back(make_hip_kernel({LOCAL_SIZE},
+                                                         {size},
+                                                         "MIOpenHingeEmbeddingLoss.cpp",
+                                                         "HingeEmbeddingLossFwd",
+                                                         build_params));
+
+    /* Phase 2: Add reduce kernels */
+    auto _size = size;
+    do
+    {
+        result.construction_params.push_back(make_hip_kernel(
+            {LOCAL_SIZE_REDUCE_FWD}, {_size}, "MIOpenLossSum.cpp", "LossSum", build_params));
+        _size = AlignUp(_size, LOCAL_SIZE_REDUCE_FWD) / LOCAL_SIZE_REDUCE_FWD;
+    } while(_size > 1);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) params = raw_params.CastTo<miopen::loss::FwdInvokeParams>();
+            auto elapsed          = 0.f;
+
+            /* Phase 1: Calc loss for each element. */
+            {
+                decltype(auto) kernel = handle_.Run(kernels.front());
+                auto input_tv         = get_inner_expanded_tv(deref(params.inputDesc));
+                auto target_tv        = get_inner_expanded_tv(deref(params.targetDesc));
+                kernel(params.input,
+                       params.target,
+                       params.workspace,
+                       params.margin,
+                       params.divisor,
+                       input_tv,
+                       target_tv);
+            }
+            if(handle_.IsProfilingEnabled())
```

Comment:
When multiple kernels are running, performance should be measured from the moment the first kernel starts to the time the last kernel finishes. Please refer to src/solver/layernorm/backward_t5layernorm.cpp.

Reply:
I have updated measurement method following your suggestion

Reply:
However, I think this approach is still unfair because: In parse_rocprof script to benchmark ROCm, metric `kernel_duration_avg` is used to compare with MIOpen. However, in this metric, the idle time between the end of one operation and the start of the next operation is not included.

Reply:
Additionally, I see others approved loss PRs such as NLLoss, SmoothL1Loss apply the same approach.